db/c_test.c

```diff
@@ -11,8 +13,13 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+using leveldb::ValueType;
+
+struct leveldb_keymetadata_t  { leveldb::KeyMetaData       rep; };
```

Comment:
This declaration breaks stated convention #1 in c.h that the objects passed to the C API are opaque struct pointers.

Perhaps we don't really care about maintaining this characteristic of the C API. Whether we do or not, should probably add a comment here stating that this declaration must match the corresponding declaration in c.cc.